pkg/pillar/types/assignableadapters.go

```diff
@@ -50,38 +60,53 @@ func (iobe *ioBundleError) String() string {
 	return strings.Join(errorStrings, "; ")
 }
 
-func (iobe *ioBundleError) Append(err error) {
+// Append converts an error to ioBundleErrorBase and adds it
+func (iobe *IOBundleError) Append(err error) {
 	if iobe.Errors == nil {
-		iobe.Errors = make([]error, 0, 1)
+		iobe.Errors = make([]ioBundleErrorBase, 0, 1)
 	}
 
-	iobe.Errors = append(iobe.Errors, err)
+	typeStr := reflect.TypeOf(err).String()
+	baseErr := ioBundleErrorBase{
+		ErrStr:  err.Error(),
+		TypeStr: typeStr,
+	}
+
+	iobe.Errors = append(iobe.Errors, baseErr)
 
 	iobe.TimeOfError = time.Now()
 }
 
-func (iobe *ioBundleError) Empty() bool {
+// Empty returns true if no error has been added
+func (iobe *IOBundleError) Empty() bool {
 	if iobe.Errors == nil || len(iobe.Errors) == 0 {
 		return true
 	}
 
 	return false
 }
 
-func (iobe *ioBundleError) hasError(e error) bool {
+// HasError returns true if error of the same type is found
+func (iobe *IOBundleError) HasError(e error) bool {
```

Comment:
I know it's unrelated to this PR, but maybe we can rename it `HasErrorType` or something similar?

Reply:
changed it to `HasErrorByType` same as `removeByType`

Reply:
Thanks